Clinical-Genomics/cg
# add raredisease pedigree check

## Description
The RareDisease requires an additional metric from the multiQC data file with respect to the other Nextflow pipelines, which is the pedigree check. This is a pair-wise sample  metric, meaning there is a value for every pair of samples combination in the case. To parse them, the method `get_multiqc_json_metrics` from `NfAnalysisAPI` is overridden in `RarediseaseAnalysisAPI`.

### Added

- Pedigree check in raredisease QC: parent_error_ped_check

### Changed

- MEDIAN_TARGET_COVERAGE threshold to >25

### Fixed

-


### How to prepare for test

    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b add-raredisease-pedigree-check
    ```

### How to test
- [x] Do nano /home/proj/stage/analysis/cases/funkitten/multiqc/multiqc_data/multiqc_data.json and MEDIAN_TARGET_COVERAGE set to 26 for all three samples
- [x] Do `cg workflow raredisease metrics-deliver funkitten` with MEDIAN_TARGET_COVERAGE set to 26.

- [x] Do  `sed -i 's/"parent_error_ped_check": "False",/"parent_error_ped_check": "True",/g' /home/proj/stage/analysis/cases/funkitten/multiqc/multiqc_data/multiqc_data.json`
- [x] Do `cg workflow raredisease metrics-deliver funkitten`

### Expected test outcome

- [x] Check that there are no errors, and command completes as expected.
![image](https://github.com/user-attachments/assets/5f9ebd60-6c0c-4bd3-8c9f-50fb95e989e0)


- [x] When setting parent error to True, the command should fail. Error needs to be propagated to trailblazer
![image](https://github.com/user-attachments/assets/d7596376-ae8b-4b0f-ae19-f9b029fda3b3)

- [x] See that the check is visible in the deliverables.yaml file:
![image](https://github.com/user-attachments/assets/f1b65879-51b9-462e-9ec0-663f0f7dbf0e)


## Review

Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a


## Implementation Plan
